pytorch_lightning/trainer/connectors/callback_connector.py

```diff
@@ -232,6 +241,35 @@ def _configure_fault_tolerance_callbacks(self):
         # don't use `log_dir` to minimize the chances of failure
         self.trainer.callbacks.append(_FaultToleranceCheckpoint(dirpath=self.trainer.default_root_dir))
 
+    def _configure_external_callbacks(self) -> None:
+        """Add external callbacks registered through entry points.
+
+        The entry points are expected to be functions returning a list of callbacks, which will be added to the Trainer
+        callback list.
+        """
+        if _PYTHON_GREATER_EQUAL_3_8_0:
+            factories = importlib.metadata.entry_points().get("pytorch_lightning.callbacks_factory", ())
+        else:
+            factories = pkg_resources.iter_entry_points("pytorch_lightning.callbacks_factory")
+
+        for factory in factories:
+            callbacks_list: List[Callback] = factory.load()()
+            if not isinstance(callbacks_list, list):
+                raise TypeError(
+                    f"The entry point '{factory.name}' returned a {type(callbacks_list)} but is expected to return"
+                    f" a list of `pytorch_lightning.callbacks.Callback`."
+                )
+            if not all(isinstance(cb, Callback) for cb in callbacks_list):
+                raise TypeError(
+                    f"The entry point '{factory.name}' is expected to return a list of callbacks, but at least one"
+                    " callack was not an instance of `pytorch_lightning.callbacks.Callback`."
+                )
```

Comment:
Reviewer: is this validation overkill?

Reply:
I think it is the right thing to do.

Reply:
I advocate for removing it, it goes against python duck typing philosophy.

But if we want it anyways, it shouldn't be done here but in the config validator or in the `on_trainer_init` method for all callbacks including those passed to the Trainer directly.

Reply:
I am not sure. As a user, I would prefer to know why it is failing. If we wait for on_trainer_init, we don't know if the callback comes from the Trainer or if it is injected as entry points.